robot/impl/resource_manager.go

```diff
@@ -470,6 +470,9 @@ func checkReconfigure(manager *resourceManager, gNode *resource.GraphNode) bool
 			manager.logger.Debug(err)
 		} else {
 			manager.logger.Error(err)
+			if resLogger := gNode.GetLogger(); resLogger != nil {
```

Comment:
I definitely noticed the pattern that I added all of these `GetLogger` calls where `gNode.SetLastError` were called. I believe @benjirewis added `SetLastError` to help surface configuration problems to the UI.

I'm open to just having `SetLastError` do the logging. The coupling seems reasonable to me. Maybe change the name to make it more obvious it logs as well? The existing name would also be fine. Curious about others thoughts here.

Reply:
> I believe @benjirewis added SetLastError to help surface configuration problems to the UI.

That, while possible, is not something I recall doing 😆 . I'm aligned on the fact that the goal of setting the last error is so that future references to the resource return the relevant configuration problem (and that problem is "surfaced" to the UI through logs). I would be in favor of logging in `SetLastError` and renaming it to something like `LogAndSetLastError`.